python/python/bystro/supervised_ppca/tests/test_dropout_pt.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+from sklearn.linear_model import LogisticRegression  # type: ignore
+from sklearn.metrics import roc_auc_score  # type: ignore
+from scipy.stats import ortho_group  # type: ignore
+from scipy.stats import binom  # type: ignore
+
+from bystro.supervised_ppca.gf_dropout_pt import PPCADropout
+
+
+def PPCA_generate_data(N=10000, L=5, p=30, phi=1.0, sigma=1.0):
+    rng = np.random.default_rng(2021)
+    S = rng.normal(size=(N, L))
+    W_o = ortho_group.rvs(p) * np.sqrt(p)
+    W = W_o[:L] * 3
+    W[0] = 0.3 * W[0]
+
+    X_hat = np.dot(S, W)
+    X_noise = sigma * rng.normal(size=(N, p))
+    X = X_hat + X_noise
+
+    logits = phi * S[:, 0] * 5
+    probs = np.exp(logits) / (1 + np.exp(logits))
+    y = binom.rvs(1, probs, random_state=42)
+    return X, y, X_hat, S, W, logits
+
+
+def test_ppca():
+    X, y, X_hat, S, W, logits = PPCA_generate_data(L=3, p=200)
+    training_options = {"n_iterations": 1000}
```

Comment:
Please be careful with these: they can lead to qualitative slow-downs in development velocity for everyone else.  If we're unit testing more than a few iterations of a pytorch model we're performing the wrong test.  

Is 1000 iterations demonstrably necessary here or would a smaller number suffice?

Reply:
That's a discussion, do we want unit tests to evaluate performance in realistic scenarios, in which case probably yes.